### PR TITLE
Time collected added to PeakData and PeakGroups advanced search formats

### LINKED ISSUE #621: Metadata in downloaded data is missing the field Time Collected
### Problem

When downloading data from the Advanced Search page, one important column is missing: the time collected for the sample.  I believe this value is `Time Collected` in the sample information sheet.

### Steps to reproduce

Download PeakGroups.tsv, PeakData.tsv, or Fcirc.tsv from the advanced search page.

### Suggested Change

Add the `Time Collected` value to the exported columns for all three download formats.

### Comment

This value is required for any comparisons made in serum samples over time (ie to plot the change in labeling or Fcirc in multiple serum samples from an infusion.

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected Components

A list of repository items, dependencies, etc, labeled with add, delete, or
change.  One item per line.  (Mostly, this will be a list of files.)

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

#### GUI Change description

Describe changes the user will see.

#### Code Change Description

Describe code changes planned for the feature. *(Pseudocode encouraged)*

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

Added `Sample.time_collected` to the PeakData and PeakGroups Advanced search formats.

- Also fixed a column header for body weight, which contained a double "(g)".

## Affected Issue Numbers

- Resolves #621

## Code Review Notes

Pretty straightforward.  The toughest part was updating the tests.

## Checklist

- [x] Changes have been added to the *Unreleased* section in the [`changelog.md`](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md).
- [x] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
  - The linting errors predate this PR. All linting of changed files in this PR pass
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- Tests
  - [ ] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
    - Pre-existing tests were updated.
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All example load tests pass (remotely) *(or their failures predated and are unrelated to this branch)*

### CODE REVIEW

File: DataRepo/formats/peakdata_dataformat.py [hunk 1]
```diff
@@ -288,6 +288,21 @@ class PeakDataFormat(Format):
...
+                    "units": {  # See dataformat.py: Format.unit_options
+                        "key": "postgres_interval",
+                        "default": "minutes",
+                        "subset": [
+                            "hours",
+                            "minutes",
+                            "seconds",
+                        ],
+                    },
+                },
```
Comment (hepcat72) [inline DataRepo/formats/peakdata_dataformat.py:305]: C. This block and the one in the peakgroups format were straight-up copied from the fcirc format, which already has this field.

File: DataRepo/templates/DataRepo/search/downloads/peakdata_row.tsv [hunk 2]
```diff
@@ -1 +1 @@
...
...msrun.sample.animal.diet }}	{{ row.peak_group.msrun.sample.animal.feeding_status }}	{% if row.peak_group.msrun.treatment.name is None %}None{% else %}{{ row.peak_group.msrun.treatment.name }}{% endif %}	{% if row.peak_group.msrun.sample.animal.infusate.short_name is None %}None{% else %}{{ row.peak_group.msrun.sample.animal.infusate.short_name }}{% endif %}	{% for link in row.peak_group.msrun.sample.animal.infusate.tracer_links.all %}{% if not forloop.first %};{% endif%}{% if link.tracer.get_name is None %}None{% else %}{{ link.tracer.get_name }}{% endif %}{% endfor %}	{% for link in row.peak_group.msrun.sample.animal.infusate.tracer_links.all %}{% if not forloop.first %};{% endif%}{% if link.tracer.compound.name is None %}None{% else %}{{ link.tracer.compound.name }}{% endif %}{% endfor %}	{% for link in row.peak_group.msrun.sample.animal.infusate.tracer_links.all %}{% if not forloop.first %},{% endif%}{% if link.concentration is None %}None{% else %}{{ link.concentration }}{% endif %}{% endfor %}	{{ row.peak_group.msrun.sample.animal.infusion_rate }}	{% for study in row.peak_group.msrun.sample.animal.studies.all %}{% if not forloop.first %}, {% endif%}{{ study.name }}{% endfor %}
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/search/downloads/peakdata_row.tsv:1]: C. Added `{{ row.peak_group.msrun.sample.time_collected|durationToMins }}`.  Not sure why it isn't highlighted in dark green.  That's the only change.  It was copied from the fcirc template and edited.

File: DataRepo/templates/DataRepo/search/downloads/peakgroups_row.tsv [hunk 3]
```diff
@@ -1 +1 @@
...
...ow.msrun.sample.animal.age|durationToWeeks }}	{{ row.msrun.sample.animal.sex }}	{{ row.msrun.sample.animal.diet }}	{{ row.msrun.sample.animal.feeding_status }}	{% if row.msrun.sample.animal.treatment is None %}None{% else %}{{ row.msrun.sample.animal.treatment.name }}{% endif %}	{% if row.msrun.sample.animal.infusate.short_name is None %}None{% else %}{{ row.msrun.sample.animal.infusate.short_name }}{% endif %}	{% for link in row.msrun.sample.animal.infusate.tracer_links.all %}{% if not forloop.first %};{% endif%}{% if link.tracer.get_name is None %}None{% else %}{{ link.tracer.get_name }}{% endif %}{% endfor %}	{% for link in row.msrun.sample.animal.infusate.tracer_links.all %}{% if not forloop.first %};{% endif%}{% if link.tracer.compound.name is None %}None{% else %}{{ link.tracer.compound.name }}{% endif %}{% endfor %}	{% for link in row.msrun.sample.animal.infusate.tracer_links.all %}{% if not forloop.first %},{% endif%}{% if link.concentration is None %}None{% else %}{{ link.concentration }}{% endif %}{% endfor %}	{{ row.msrun.sample.animal.infusion_rate }}	{% for study in row.msrun.sample.animal.studies.all %}{% if not forloop.first %}, {% endif%}{{ study.name }}{% endfor %}
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/search/downloads/peakgroups_row.tsv:1]: C. Added `{{ row.msrun.sample.time_collected|durationToMins }}`.  Not sure why it isn't highlighted in dark green.  That's the only change.  It was copied from the fcirc template and edited.

File: DataRepo/templates/DataRepo/search/results/peakdata.html [hunk 4]
```diff
@@ -185,7 +192,7 @@
                 <th data-valign="top" class="metagrp" data-visible="{% get_template_cookie selfmt 'Body_Weight-data-visible' 'false' %}" data-field="Body_Weight">
                     {% with "Body<br>Weight<br>(g)" as colhead %}
                         {% if mode == "view" %}{{ colhead }}{% else %}
-                            <div onclick="sortColumn(this)" class="sortable" id="peak_group__msrun__sample__animal__body_weight">{{ colhead }}<br>(g)</div>
+                            <div onclick="sortColumn(this)" class="sortable" id="peak_group__msrun__sample__animal__body_weight">{{ colhead }}</div>
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/search/results/peakdata.html:195]: C. This is the only unrelated change.  I noticed that the header appeared as "Body Weight (g)(g)", so I removed the unintended second "(g)"

File: DataRepo/tests/test_formats.py [hunk 5]
```diff
@@ -305,6 +314,33 @@ def assertIsAPgUnitsLookupDict(self, fld_units_lookup):
...
+            "1w,1d,1:01:01.1",
+            fld_units_lookup["msrun__sample__time_collected"]["identity"]["example"],
+        )
+        # The convert key should be a function
+        self.assertEqual(
+            "function",
+            type(
+                fld_units_lookup["msrun__sample__time_collected"]["identity"]["convert"]
+            ).__name__,
+        )
```
Comment (hepcat72) [inline DataRepo/tests/test_formats.py:342]: C. This was the most complex part of the change.  The data structure unfortunately cannot be directly compared because there is at least 1 value that is never the same, so all these tests are indirect.

File: DataRepo/tests/test_formats.py [hunk 6]
```diff
@@ -474,6 +510,7 @@ def test_getTrueJoinPrefetchPathsAndQrys(self):
         self.assertEqual(expected_prefetches[0:4], prefetches[0:4])
         self.assertEqual(expected_prefetches[5:3], prefetches[5:3])
         self.assertEqual(expected_prefetches[4][0:3], prefetches[4][0:3])
+        print(f"PREFETCHES: {prefetches}")
```
Comment (hepcat72) [inline DataRepo/tests/test_formats.py:513]: C. Whoops. This will be removed momentarily.

File: DataRepo/templates/DataRepo/search/downloads/peakgroups_row.tsv (hunk 3 above)
Comment (lparsons) [inline DataRepo/templates/DataRepo/search/downloads/peakgroups_row.tsv:1]: Not really related to this PR, but I wonder if we could use the [`spaceless` template tag](https://docs.djangoproject.com/en/3.2/ref/templates/builtins/#spaceless) to make this more readable. Not really sure, but it might help some.

**Review by fkang-pu — APPROVED** [review]

**Review by lparsons — APPROVED** [review]
LGTM

File: DataRepo/templates/DataRepo/search/downloads/peakgroups_row.tsv (hunk 3 above)
Comment (hepcat72) [reply DataRepo/templates/DataRepo/search/downloads/peakgroups_row.tsv:1]: Yeah, it's difficult to edit too.
